taf/tests/test_updater/test_update/test_update_invalid.py

```diff
@@ -73,3 +79,186 @@ def test_update_invalid_repo_target_in_indeterminate_state(
         UNCOIMITTED_CHANGES,
         True,
     )
+
+
+@pytest.mark.parametrize(
+    "origin_auth_repo",
+    [
+        {
+            "targets_config": [{"name": "target1"}, {"name": "target2"}],
+        },
+    ],
+    indirect=True,
+)
+def test_dirty_index_auth_repo_update_file(origin_auth_repo, client_dir):
+    clone_repositories(
+        origin_auth_repo,
+        client_dir,
+    )
+
+    clients_auth_repo_path = client_dir / origin_auth_repo.name
+    update_file_without_commit(str(clients_auth_repo_path), "dirty_file.txt")
+
+    try:
+        # Attempt to update, which should fail due to uncommitted changes
+        update_and_check_commit_shas(
```

Comment:
This looks good now, the update should fail. Instead of calling this with `try-except`, we want to call `update_invalid_repos_and_check_if_repos_exist` because you can specify the error that we expect to be raised.  E.g.
```
update_invalid_repos_and_check_if_repos_exist(
        OperationType.UPDATE,
        origin_auth_repo,
        client_dir,
        TARGET_MISSMATCH_PATTERN,
        True,
    )
``` 
Here we expect `TARGET_MISSMATCH_PATTERN = r"Update of (\w+)\/(\w+) failed due to error: Failure to validate (\w+)\/(\w+) commit ([0-9a-f]{40}) committed on (\d{4}-\d{2}-\d{2}): data repository ([\w\/-]+) was supposed to be at commit ([0-9a-z]{40}) but repo was at ([0-9a-f]{40})"`
This is this function includes:

```
    def _update_expect_error():
        with pytest.raises(UpdateFailedError, match=expected_error):
            if operation == OperationType.CLONE:
                clone_repository(config)
            else:
                update_repository(config)
```
So, we want to use `with pytest.raises`
Let me know if that function does not do the trick

Reply:
I get this error when running update_invalid_repos_and_check_if_repos_exist


E       AssertionError: Regex pattern did not match.
E        Regex: 'Update of (\\w+)\\/(\\w+) failed due to error: Failure to validate (\\w+)\\/(\\w+) commit ([0-9a-f]{40}) committed on (\\d{4}-\\d{2}-\\d{2}): data repository ([\\w\\/-]+) was supposed to be at commit ([0-9a-z]{40}) but repo was at ([0-9a-f]{40})'
E        Input: 'Update of repository failed due to error: Repository test_dirty_index_auth_repo_update_file0/auth has uncommitted changes. Commit and push or revert the changes and run the command again.'